### ISSUE TITLE
flux command driver eats subcommand options

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#531]
Running the broker via the flux command driver, broker options seem to be parsed by the flux command driver:
```
$ ./flux broker --nonoption
flux broker: unrecognized option '--nonoption'
Try `flux broker --help' for more information.
```

Here is the broker failing to parse the same option:
```
$ ../broker/flux-broker --nonoption
/home/garlick/proj/flux-core/src/broker/.libs/lt-flux-broker: unrecognized option '--nonoption'
Usage: flux-broker OPTIONS [module:key=val ...]
 -v,--verbose                 Be annoyingly verbose
 -q,--quiet                   Be mysteriously taciturn
 -N,--sid NAME                Set session id
 -M,--module NAME             Load module NAME (may be repeated)
 -x,--exclude NAME            Exclude module NAME
 -O,--modopt NAME:key=val     Set option for module NAME (may be repeated)
 -X,--module-path PATH        Set module search path (colon separated)
 -L,--logdest FILE            Redirect log output to specified file
 -l,--log-level LEVEL         Set log level (default: 6/info)
          [0=emerg 1=alert 2=crit 3=err 4=warning 5=notice 6=info 7=debug]
 -s,--security=plain|curve|none    Select security mode (default: curve)
 -k,--k-ary K                 Wire up in a k-ary tree
 -H,--heartrate SECS          Set heartrate in seconds (rank 0 only)
 -T,--timeout SECS            Set wireup timeout in seconds (rank 0 only)
 -g,--shutdown-grace SECS     Set shutdown grace period in seconds
 -D,--scratch-directory DIR   Scratch directory for sockets, etc
 -E,--enable-epgm             Enable EPGM for events (PMI bootstrap)
 -I,--shared-ipc-namespace    Wire up session TBON over ipc sockets
 -m,--boot-method             Select bootstrap: pmi, single
$
```

**Comment by grondo** [comment]
Ah, oops! `flux broker` is a special case of a subcommand where we don't actually want to do automatic option processing, because all it does is exec another utility, passing along options. Let me look at this we might need a new option for optparse subcommands to handle this case

(And I'm embarrassed I missed this in the the earlier work)

**Comment by grondo** [comment]
Hm, all `flux broker` does is set `PATH` to `flux-broker` path and then `execvp()`.
Is there some reason we did not move `flux-broker` into `FLUX_CMD_PATH` so it could be found by default? Then we don't need the extra level of indirection of a builtin `flux broker` calling `flux-broker`?

I'm fine with fixing the optparse issue too, and I'm probably missing the purpose of putting the flux-broker in its own path...

**Comment by garlick** [comment]
> Is there some reason we did not move flux-broker into FLUX_CMD_PATH so it could be found by default? 

Sounds fine with me.  Do you mean install flux-broker to `$(fluxcmddir)`?  How does that work in tree?

**Comment by grondo** [comment]
Good question. Could we just add `FLUX_BROKER_PATH` to exec-path?

**Comment by garlick** [comment]
Seems like that would do it.  I'll propose something on #526 unless you've already got something going.

**Comment by grondo** [comment]
I started on something, but what is giving me pause is that this would apparently make it more difficult to run an alternate flux-broker, or flux-broker from another installation under a given `flux` command. I'm not sure if that is a problem or if I'm worried over nothing..

**Comment by grondo** [comment]
You can propose something in #526 if you'd like. I'll concentrate of fixing the optparse bug specifically described here.

**Comment by garlick** [comment]
> this would apparently make it more difficult to run an alternate flux-broker, or flux-broker from another installation under a given flux command.

Woudn't `flux --execpath` cover that use case?

**Comment by grondo** [comment]
> Woudn't flux --execpath cover that use case?

Yeah, good point. Or just `flux /path/to/my/flux-broker ...`

**Comment by grondo** [comment]
Both  the optparse issue and `flux broker` cmdline issue described should be fixed in master now with the merging of #532 and #526
